src/plugins/wizard/public/application/utils/get_top_nav_config.tsx

```diff
@@ -130,6 +139,24 @@ export const getTopNavConfig = (
                 'data-test-subj': 'saveVisualizationSuccess',
               });
 
+              if (originatingApp && returnToOrigin) {
```

Comment:
Since this function seems to have a lot of application logic, can we add some unit tests now to ensure that all the logic in this function behaves as expected (not just for the functionality that you have added but also the rest)?

Reply:
Break down the large topNavConfig() function into three sub functions, and added unit tests. Added a createWizardServicesMock() for wizard plugin.